File: compiler/src/main/java/com/aaroncoplan/waterfall/compiler/statements/helpers/VerificationResult.java
package com.aaroncoplan.waterfall.compiler.statements.helpers;

import com.aaroncoplan.waterfall.parser.Pair;

public class VerificationResult extends Pair<Boolean, String> {

    public VerificationResult(boolean isSuccessful, String errorMessage) {
        super(isSuccessful, errorMessage);
    }

    public boolean isSuccessful() {
        return this.firstVal;
    }

    public String getErrorMessage() {
        return this.secondVal;
    }
}

File: compiler/src/main/kotlin/com/aaroncoplan/waterfall/compiler/statements/helpers/VerificationResult.kt
package com.aaroncoplan.waterfall.compiler.statements.helpers

import com.aaroncoplan.waterfall.parser.Pair

class VerificationResult(isSuccessful: Boolean, errorMessage: String?)
    : Pair<Boolean, String?>(isSuccessful, errorMessage) {

    fun isSuccessful(): Boolean = firstVal
    fun getErrorMessage(): String? = secondVal
}
